avro/src/main/java/com/fasterxml/jackson/dataformat/avro/AvroAnnotationIntrospector.java

```diff
@@ -0,0 +1,238 @@
+package com.fasterxml.jackson.dataformat.avro;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.introspect.Annotated;
+import com.fasterxml.jackson.databind.introspect.JacksonAnnotationIntrospector;
+import com.fasterxml.jackson.dataformat.avro.deser.AvroParserImpl;
+import org.apache.avro.Schema;
+import org.apache.avro.io.Decoder;
+import org.apache.avro.io.Encoder;
+import org.apache.avro.reflect.AvroEncode;
+import org.apache.avro.reflect.CustomEncoding;
+import org.apache.avro.util.Utf8;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.ByteBuffer;
+
+public class AvroAnnotationIntrospector extends JacksonAnnotationIntrospector {
+    public static class AvroDeserializer<T> extends JsonDeserializer<T> {

```

Comment:
I probably will not have much chance to update the PR for the next few days, but will go back and add documentation when I do; These are internal classes that wrap an avro `CustomEncoding` and allow it to be registered into Jackson as a standard `JsonDeserializer`. I'm still working on the counterpart that functions as a `JsonSerializer`. This is to suppor the `@AvroEncode` annotation.